src/target/source/ptx_mma.cc

```diff
@@ -0,0 +1,1362 @@
+  return asm_code;
+}
+
+std::string PrintPTXAssembly(const std::string& shape, const std::string& A_layout,
+                             const std::string& B_layout, const std::string& A_dtype,
+                             const std::string& B_dtype, const std::string& C_dtype,
+                             const std::string& a_ref, const std::string& a_bias,
+                             const std::string& b_ref, const std::string& b_bias,
+                             const std::string& c_ref, const std::string& c_bias, bool saturate) {
+  ICHECK((shape == "m8n8k4") || (shape == "m16n8k8") || (shape == "m8n8k16") ||
+         (shape == "m8n8k32") || (shape == "m16n8k4") || (shape == "m16n8k16") ||
+         (shape == "m16n8k32") || (shape == "m16n8k64") || (shape == "m16n8k256"));
+  ICHECK((A_layout == "row") || (A_layout == "col")) << "Unknown A layout: " << A_layout;
+  ICHECK((B_layout == "row") || (B_layout == "col")) << "Unknown B layout: " << B_layout;
+
+  if (shape == "m8n8k4") {
+    return PrintMMAm8n8k4Assembly(A_layout, B_layout, A_dtype, B_dtype, C_dtype, a_ref, a_bias,
+                                  b_ref, b_bias, c_ref, c_bias, saturate);
+  } else if (shape == "m16n8k8") {
+    return PrintMMAm16n8k8Assembly(A_layout, B_layout, A_dtype, B_dtype, C_dtype, a_ref, a_bias,
+                                   b_ref, b_bias, c_ref, c_bias, saturate);
+  } else if (shape == "m8n8k16") {
+    return PrintMMAm8n8k16Assembly(A_layout, B_layout, A_dtype, B_dtype, C_dtype, a_ref, a_bias,
+                                   b_ref, b_bias, c_ref, c_bias, saturate);
+  } else if (shape == "m8n8k32") {
+    return PrintMMAm8n8k32Assembly(A_layout, B_layout, A_dtype, B_dtype, C_dtype, a_ref, a_bias,
+                                   b_ref, b_bias, c_ref, c_bias, saturate);
+  } else if (shape == "m16n8k4") {
+    return PrintMMAm16n8k4Assembly(A_layout, B_layout, A_dtype, B_dtype, C_dtype, a_ref, a_bias,
+                                   b_ref, b_bias, c_ref, c_bias, saturate);
+  } else if (shape == "m16n8k16") {
+    return PrintMMAm16n8k16Assembly(A_layout, B_layout, A_dtype, B_dtype, C_dtype, a_ref, a_bias,
+                                    b_ref, b_bias, c_ref, c_bias, saturate);
+  } else if (shape == "m16n8k32") {
+    return PrintMMAm16n8k32Assembly(A_layout, B_layout, A_dtype, B_dtype, C_dtype, a_ref, a_bias,
+                                    b_ref, b_bias, c_ref, c_bias, saturate);
+  } else if (shape == "m16n8k64") {
+    return PrintMMAm16n8k64Assembly(A_layout, B_layout, A_dtype, B_dtype, C_dtype, a_ref, a_bias,
+                                    b_ref, b_bias, c_ref, c_bias, saturate);
+  } else if (shape == "m16n8k256") {
+    return PrintMMAm16n8k256Assembly(A_layout, B_layout, A_dtype, B_dtype, C_dtype, a_ref, a_bias,
+                                     b_ref, b_bias, c_ref, c_bias, saturate);
+  }
+  /*
+   * TODO: add mma.m16n8k128
+   */
+  return "";
```

Comment:
```suggestion
  ICHECK(0);
  throw;
```
if this is unreachable, just raises an error